Mulitple memory server - services initialization

faizan-barmawer opened the pull request:
- All services read options from the configuration file.
  - Added get_config_info methods in *_Direct classes to read
    options from config file
  - Initialization sequence for various combination of services.

All combination of different services needs to be tested.

rajakrishi commented:
Can one of the admins verify this patch?

mashoodk commented on src/cis/fam_cis_direct.cpp:
```diff
@@ -873,4 +919,61 @@ void Fam_CIS_Direct::get_addr(void *memServerFabricAddr,
+            for (auto item : temp)
+                metasrvList << item << ",";
+
+            options["metadata_list"] = metasrvList.str().c_str();
```
Clarification : There is a parameter, rpc_interface, how are we incorporating that ? Or you think we need not keep it.

faizan-barmawer commented:
Addressed review comments from Mashood and updated a new commit.

Regards,
Faizan

faizan-barmawer commented:
Addressed additional review comments. 
Retained only one build for memoryserver model. Added test runs for different service configuration in build_and_test script.

Regards,
Faizan

gautham-bhat-k commented on build_and_test.sh:
```diff
@@ -1,105 +1,144 @@
 #!/bin/bash
 CURRENTDIR=`pwd`
-CIS_RPC_BUILD_DIR=$CURRENTDIR/build/build-cis-rpc
+RPC_BUILD_DIR=$CURRENTDIR/build/build-rpc
```
Will it be appropriate to name build-mem-server or similar name ?

faizan-barmawer replied:
Renamed the build directory for memory server model and pushed a new commit.

Regards,
Faizan

gautham-bhat-k commented on build_and_test.sh:
```diff
@@ -1,105 +1,144 @@
-if [ ! -d "$CIS_RPC_BUILD_DIR" ]; then
-  echo "Creating $CIS_RPC_BUILD_DIR"
-  mkdir -p $CIS_RPC_BUILD_DIR
+if [ ! -d "$RPC_BUILD_DIR" ]; then
```
Instead of RPC_BUILD_DIR can it be MEMSERV_BUILD_DIR ?
